test/integration/puppetlabs/services/master/environment_transports_int_test.clj

```diff
@@ -0,0 +1,638 @@
+         (bootstrap/load-dev-config-with-overrides)
+         (ks/dissoc-in [:jruby-puppet
+                        :environment-class-cache-enabled]))
+     (let [foo-file (testutils/write-foo-pp-file
+                     "some kind of test data")
+           expected-response (json/encode mock-response)
+           response (get-env-transports "production")]
+       (testing "a successful status code is returned"
+         (is (= 200 (:status response))
+             (str
+              "unexpected status code for response, response: "
+              (ks/pprint-to-string response))))
+       (testing "no etag is returned"
+         (is (false? (contains? (:headers response) "etag"))))
+       (testing "the expected response body is returned"
+         (is (= expected-response
+                (response->map response))))))))
+
+
+(deftest ^:integration environment-transports-integration-cache-enabled-test1
+  (let [debug-log "./target/environment-transports-integration-cache-enabled.log"]
+    (fs/delete debug-log)
+    (bootstrap/with-puppetserver-running app
+     {:global {:logging-config
+               (str "./dev-resources/puppetlabs/services/"
+                    "master/environment_transports_int_test/"
+                    "logback-environment-transports-integration-cache-enabled-test.xml")}
+      :jruby-puppet {:gem-path gem-path
+                     :max-active-instances 1
+                     :environment-class-cache-enabled true}}
+     (try
+       (let [foo-file (str "tbd test data")
+             expected-initial-response (json/encode mock-response)
+             initial-response (get-env-transports "production")
+             initial-etag (response-etag initial-response)
+             initial-etag-with-gzip-suffix (etag-with-gzip-suffix initial-etag)
+             initial-etag-without-gzip-suffix (etag-without-gzip-suffix initial-etag)]
+         (testing "environment transport cache invalidation for one environment"
+           ;; This test is about ensuring that when the environment-cache
+           ;; endpoint is hit for a single environment that only that the
+           ;; environment class info cached for that environment - but not the
+           ;; info cached for other environments - is invalidated, meaning that
+           ;; the next request for class info for that environment will get fresh
+           ;; data.
+           ;;
+           ;; The test has the following basic steps:
+           ;;
+           ;; 1) Purge the current environment files on disk and hit the
+           ;;    environment-cache endpoint to flush the cache for all
+           ;;    environments, basically to ensure nothing is left around from
+           ;;    other tests.
+           ;; 2) Populate code for the 'test' and 'production' environments and
+           ;;    see that environment_classes queries return the right info for
+           ;;    them.
+           ;; 3) Do one more environment_classes query for each environment,
```

Comment:
I think this comment should be referring to `environment_transports` throughout?